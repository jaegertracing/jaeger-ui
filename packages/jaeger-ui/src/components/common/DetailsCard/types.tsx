// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
import type React from 'react';
import { TExample } from '../ExamplesLink';

export type TStyledValue = {
  linkTo?: string;
  styling?: React.CSSProperties;
  value: string | React.ReactElement;
};

export type TColumnDef = {
  key: string;
  label?: string;
  preventSort?: boolean;
  styling?: React.CSSProperties;
};

export type TColumnDefs = (string | TColumnDef)[];

export type TRow = Record<string, TStyledValue | string | number | TExample[]>;

export type TDetails = string | string[] | TRow[];

export type TFilterDropdownProps = {
  clearFilters?: () => void;
  confirm: () => void;
  selectedKeys: React.Key[];
  setSelectedKeys: (selectedKeys: React.Key[]) => void;
};
