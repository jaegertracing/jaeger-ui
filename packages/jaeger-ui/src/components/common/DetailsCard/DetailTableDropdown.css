/*
Copyright (c) 2019 Uber Technologies, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.DetailTableDropdown--Footer {
  background: white;
  border: 1px solid #ccc;
  display: flex;
  justify-content: space-between;
  padding: 0.3em;
}

.DetailTableDropdown--Footer--CancelConfirm {
  display: flex;
}

.DetailTableDropdown--Btn {
  align-items: center;
  border-radius: 0;
  color: #ddd;
  display: flex;
  padding: 0.3em 0.7em 0.3em 0.6em;
}

.DetailTableDropdown--Btn ~ .DetailTableDropdown--Btn {
  margin-left: 0.3em;
}

.DetailTableDropdown--Btn > *:not(:first-child) {
  margin-left: 0.3em;
}

.DetailTableDropdown--Btn.Apply {
  background: #4c21ce;
}

.DetailTableDropdown--Btn.Cancel {
  background: #007272;
}

.DetailTableDropdown--Btn.Clear {
  background: #ff2626;
}

.DetailTableDropdown--Tooltip {
  max-width: unset;
  white-space: nowrap;
}

.DetailTableDropdown--Tooltip--Body {
  display: flex;
  flex-direction: column;
}
