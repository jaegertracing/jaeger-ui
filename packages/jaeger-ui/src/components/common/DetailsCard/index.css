/*
Copyright (c) 2020 The Jaeger Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.DetailsCard {
  display: flex;
  flex-direction: column;
  padding: 0.5em;
}

.DetailsCard--ButtonHeaderWrapper {
  display: flex;
}

.DetailsCard--Collapser {
  border: none;
  background: transparent;
  cursor: pointer;
  flex: 0;
  font-size: 1.5rem;
  transition: transform 0.25s ease-out;
}

.DetailsCard--Collapser:focus {
  outline: none;
}

.DetailsCard--Collapser.is-collapsed {
  transform: rotate(-90deg);
}

.DetailsCard--HeaderWrapper {
  flex-grow: 1;
}

.DetailsCard--Header {
  border-bottom: solid 1px #ddd;
  box-shadow: 0px 5px 5px -5px rgba(0, 0, 0, 0.3);
  font-size: 1.5em;
  padding: 0.1em 0.3em;
}

.DetailsCard--Description {
  margin-bottom: 0;
  max-width: 90%;
}

.DetailsCard--DetailsWrapper {
  overflow: scroll;
  margin-top: 0.5em;
  transition: max-height 0.25s ease-out;
  max-height: 60vh;
}

.DetailsCard--DetailsWrapper.is-collapsed {
  max-height: 0px;
}

.DetailsCard--DetailsWrapper th {
  min-width: 65px;
}
