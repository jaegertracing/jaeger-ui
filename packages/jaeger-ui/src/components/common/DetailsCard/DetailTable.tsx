// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { Icon, Table } from 'antd';
import FaFilter from 'react-icons/lib/fa/filter.js';
import _isEmpty from 'lodash/isEmpty';

import ExamplesLink, { TExample } from '../ExamplesLink';
import DetailTableDropdown from './DetailTableDropdown';

import { TColumnDef, TColumnDefs, TFilterDropdownProps, TRow, TStyledValue } from './types';

// exported for tests
export const _makeFilterDropdown = (dataIndex: string, options: Set<string>) => (
  props: TFilterDropdownProps
) => {
  return <DetailTableDropdown {...props} key={dataIndex} options={options} />;
};

// exported for tests
export const _onCell = (dataIndex: string) => (row: TRow) => {
  const cellData = row[dataIndex];
  if (!cellData || typeof cellData !== 'object' || Array.isArray(cellData)) return null;
  const { styling } = cellData;
  if (_isEmpty(styling)) return null;
  return {
    style: styling,
  };
};

// exported for tests
export const _onFilter = (dataIndex: string) => (value: string, row: TRow) => {
  const data = row[dataIndex];
  if (typeof data === 'object' && !Array.isArray(data) && typeof data.value === 'string') {
    return data.value === value;
  }
  return data === value;
};

// exported for tests
export const _renderCell = (cellData: undefined | string | TStyledValue) => {
  if (!cellData || typeof cellData !== 'object') return cellData;
  if (Array.isArray(cellData)) return <ExamplesLink examples={cellData} />;
  if (!cellData.linkTo) return cellData.value;
  return (
    <a href={cellData.linkTo} target="_blank" rel="noopener noreferrer">
      {cellData.value}
    </a>
  );
};

// exported for tests
export const _sort = (dataIndex: string) => (a: TRow, b: TRow) => {
  const aData = a[dataIndex];
  let aValue;
  if (Array.isArray(aData)) aValue = aData.length;
  else if (typeof aData === 'object' && typeof aData.value === 'string') aValue = aData.value;
  else aValue = aData;

  const bData = b[dataIndex];
  let bValue;
  if (Array.isArray(bData)) bValue = bData.length;
  else if (typeof bData === 'object' && typeof bData.value === 'string') bValue = bData.value;
  else bValue = bData;

  if (aValue < bValue) return -1;
  return bValue < aValue ? 1 : 0;
};

// exported for tests
export const _makeColumns = ({ defs, rows }: { defs: TColumnDefs; rows: TRow[] }) =>
  defs.map((def: TColumnDef | string) => {
    let dataIndex: string;
    let key: string;
    let sortable: boolean = true;
    let style: React.CSSProperties | undefined;
    let title: string;
    if (typeof def === 'string') {
      // eslint-disable-next-line no-multi-assign
      key = title = dataIndex = def;
    } else {
      // eslint-disable-next-line no-multi-assign
      key = title = dataIndex = def.key;
      if (def.label) title = def.label;
      if (def.styling) style = def.styling;
      if (def.preventSort) sortable = false;
    }

    const options = new Set<string>();
    rows.forEach(row => {
      const value = row[dataIndex];
      if (typeof value === 'string' && value) options.add(value);
      else if (typeof value === 'object' && !Array.isArray(value) && typeof value.value === 'string') {
        options.add(value.value);
      }
    });

    return {
      dataIndex,
      key,
      title,
      filterDropdown: Boolean(options.size) && _makeFilterDropdown(dataIndex, options),
      filterIcon: (filtered: boolean) => {
        if (filtered) return <FaFilter />;
        return <Icon type="filter" />;
      },
      onCell: _onCell(dataIndex),
      onHeaderCell: () => ({
        style,
      }),
      onFilter: _onFilter(dataIndex),
      render: _renderCell,
      sorter: sortable && _sort(dataIndex),
    };
  });

// exported for tests
export const _rowKey = (row: TRow) =>
  JSON.stringify(row, function replacer(
    key: string,
    value: TRow | undefined | string | number | TStyledValue | TExample[]
  ) {
    function isRow(v: typeof value): v is TRow {
      return v === row;
    }
    if (isRow(value)) return value;
    if (Array.isArray(value)) return JSON.stringify(value);
    if (typeof value === 'object') {
      if (typeof value.value === 'string') return JSON.stringify(value);
      return value.value.key || 'Unknown';
    }
    return value;
  });

export default function DetailTable({
  columnDefs: _columnDefs,
  details,
}: {
  columnDefs?: TColumnDefs;
  details: TRow[];
}) {
  const columnDefs: TColumnDefs = _columnDefs ? _columnDefs.slice() : [];
  const knownColumns = new Set(
    columnDefs.map(keyOrObj => {
      if (typeof keyOrObj === 'string') return keyOrObj;
      return keyOrObj.key;
    })
  );
  details.forEach(row => {
    Object.keys(row).forEach((col: string) => {
      if (!knownColumns.has(col)) {
        knownColumns.add(col);
        columnDefs.push(col);
      }
    });
  });

  return (
    <Table
      key="table"
      size="middle"
      columns={_makeColumns({ defs: columnDefs, rows: details })}
      dataSource={details}
      pagination={false}
      rowKey={_rowKey}
    />
  );
}
