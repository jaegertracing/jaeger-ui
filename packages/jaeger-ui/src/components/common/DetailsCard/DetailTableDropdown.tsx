// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { Button, Tooltip } from 'antd';
import FaCheck from 'react-icons/lib/fa/check.js';
import FaTrash from 'react-icons/lib/fa/trash.js';
import TiCancel from 'react-icons/lib/ti/cancel.js';

import FilteredList from '../FilteredList';

import { TFilterDropdownProps } from './types';

import './DetailTableDropdown.css';

type TProps = TFilterDropdownProps & {
  options: Set<string>;
};

export default class DetailTableDropdown extends React.PureComponent<TProps> {
  cancelled = false;
  selected: React.Key[] = [];

  componentDidUpdate(prevProps: TProps) {
    const { confirm, selectedKeys } = this.props;

    // If the entries in selectedKeys is unchanged, the dropdown has opened or closed.
    // Record the selectedKeys at this time for future cancellations.
    if (selectedKeys.length === prevProps.selectedKeys.length) {
      const prevKeys = new Set(prevProps.selectedKeys);
      if (selectedKeys.every(key => prevKeys.has(key))) {
        this.selected = selectedKeys;
      }
    }

    // Unfortunately antd requires setSelectedKeys and confirm to be called in different cycles.
    if (this.cancelled) {
      this.cancelled = false;
      confirm();
    }
  }

  cancel = () => {
    // Unfortunately antd requires setSelectedKeys and confirm to be called in different cycles.
    this.cancelled = true;
    this.props.setSelectedKeys(this.selected);
  };

  render() {
    const { clearFilters = () => {}, confirm, options, selectedKeys, setSelectedKeys } = this.props;

    const value = new Set<string>();
    selectedKeys.forEach(selected => {
      if (typeof selected === 'string') value.add(selected);
    });

    return (
      <div>
        <FilteredList
          addValues={(values: string[]) => {
            setSelectedKeys([...selectedKeys, ...values]);
          }}
          multi
          options={Array.from(options)}
          removeValues={(values: string[]) => {
            const remove = new Set<React.Key>(values);
            setSelectedKeys(selectedKeys.filter(key => !remove.has(key)));
          }}
          setValue={(v: string) => {
            setSelectedKeys([v]);
          }}
          value={value}
        />
        <div className="DetailTableDropdown--Footer">
          <Tooltip overlayClassName="DetailTableDropdown--Tooltip" title="Remove filter from this column">
            <Button className="DetailTableDropdown--Btn Clear" onClick={clearFilters}>
              <FaTrash size={18} />
              Clear Filter
            </Button>
          </Tooltip>
          <div className="DetailTableDropdown--Footer--CancelConfirm">
            <Tooltip
              overlayClassName="DetailTableDropdown--Tooltip"
              title="Cancel changes to this column's filter"
            >
              <Button className="DetailTableDropdown--Btn Cancel" onClick={this.cancel}>
                <TiCancel size={20} />
                Cancel
              </Button>
            </Tooltip>
            <Tooltip
              overlayClassName="DetailTableDropdown--Tooltip"
              title={
                <div className="DetailTableDropdown--Tooltip--Body">
                  <span>Apply changes to this column{"'"}s filter</span>
                  <span>Same effect as clicking outside the dropdown</span>
                </div>
              }
            >
              <Button className="DetailTableDropdown--Btn Apply" onClick={confirm}>
                <FaCheck size={18} />
                Apply
              </Button>
            </Tooltip>
          </div>
        </div>
      </div>
    );
  }
}
