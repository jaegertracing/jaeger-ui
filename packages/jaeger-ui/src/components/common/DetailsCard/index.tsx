// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import cx from 'classnames';
import { MdKeyboardArrowDown } from 'react-icons/md';

import { TColumnDefs, TDetails, TRow } from './types';
import DetailTable from './DetailTable';
import DetailList from './DetailList';

import './index.css';

type TProps = {
  className?: string;
  collapsible?: boolean;
  columnDefs?: TColumnDefs;
  description?: string;
  details: TDetails;
  header: string;
};

type TState = {
  collapsed: boolean;
};

function isList(arr: string[] | TRow[]): arr is string[] {
  return typeof arr[0] === 'string';
}

export default class DetailsCard extends React.PureComponent<TProps> {
  state: TState;

  constructor(props: TProps) {
    super(props);

    this.state = { collapsed: Boolean(props.collapsible) };
  }

  renderDetails() {
    const { columnDefs, details } = this.props;

    if (Array.isArray(details)) {
      if (details.length === 0) return null;

      if (isList(details)) return <DetailList details={details} />;
      return <DetailTable columnDefs={columnDefs} details={details} />;
    }

    return <span>{details}</span>;
  }

  toggleCollapse = () => {
    this.setState((prevState: TState) => ({
      collapsed: !prevState.collapsed,
    }));
  };

  render() {
    const { collapsed } = this.state;
    const { className, collapsible, description, header } = this.props;

    return (
      <div className={cx('DetailsCard', className)}>
        <div className="DetailsCard--ButtonHeaderWrapper">
          {collapsible && (
            <button
              onClick={this.toggleCollapse}
              type="button"
              className={cx('DetailsCard--Collapser', { 'is-collapsed': collapsed })}
            >
              <MdKeyboardArrowDown />
            </button>
          )}
          <div className="DetailsCard--HeaderWrapper">
            <span className="DetailsCard--Header">{header}</span>
            {description && <p className="DetailsCard--Description">{description}</p>}
          </div>
        </div>
        <div className={cx('DetailsCard--DetailsWrapper', { 'is-collapsed': collapsed })}>
          {this.renderDetails()}
        </div>
      </div>
    );
  }
}
