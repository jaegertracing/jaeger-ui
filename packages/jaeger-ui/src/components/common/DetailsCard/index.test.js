// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import { shallow } from 'enzyme';

import DetailsCard from '.';

describe('DetailsCard', () => {
  const header = 'Details Card Header';

  it('renders string details', () => {
    const details = 'test details';
    expect(shallow(<DetailsCard details={details} header={header} />)).toMatchSnapshot();
  });

  it('handles empty details array', () => {
    const details = [];
    expect(shallow(<DetailsCard details={details} header={header} />)).toMatchSnapshot();
  });

  it('renders list details', () => {
    const details = ['foo', 'bar', 'baz'];
    expect(shallow(<DetailsCard details={details} header={header} />)).toMatchSnapshot();
  });

  it('renders table details', () => {
    const details = [{ value: 'foo' }];
    expect(shallow(<DetailsCard details={details} header={header} />)).toMatchSnapshot();
  });

  it('renders table details with column defs', () => {
    const columnDefs = ['col'];
    const details = [{ [columnDefs[0]]: 'foo' }];
    expect(
      shallow(<DetailsCard columnDefs={columnDefs} details={details} header={header} />)
    ).toMatchSnapshot();
  });

  it('renders with description', () => {
    const description = 'test description';
    expect(shallow(<DetailsCard description={description} header={header} />)).toMatchSnapshot();
  });

  it('renders with className', () => {
    const className = 'test className';
    expect(shallow(<DetailsCard className={className} header={header} />)).toMatchSnapshot();
  });

  it('renders as collapsible', () => {
    expect(shallow(<DetailsCard header={header} />).state('collapsed')).toBe(false);

    const wrapper = shallow(<DetailsCard collapsible header={header} />);
    expect(wrapper.state('collapsed')).toBe(true);
    expect(wrapper).toMatchSnapshot();

    wrapper.find('button').simulate('click');
    expect(wrapper.state('collapsed')).toBe(false);

    wrapper.find('button').simulate('click');
    expect(wrapper.state('collapsed')).toBe(true);
  });
});
