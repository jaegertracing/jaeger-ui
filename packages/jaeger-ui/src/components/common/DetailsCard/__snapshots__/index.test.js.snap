// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsCard handles empty details array 1`] = `
<div
  className="DetailsCard"
>
  <div
    className="DetailsCard--ButtonHeaderWrapper"
  >
    <div
      className="DetailsCard--HeaderWrapper"
    >
      <span
        className="DetailsCard--Header"
      >
        Details Card Header
      </span>
    </div>
  </div>
  <div
    className="DetailsCard--DetailsWrapper"
  />
</div>
`;

exports[`DetailsCard renders as collapsible 1`] = `
<div
  className="DetailsCard"
>
  <div
    className="DetailsCard--ButtonHeaderWrapper"
  >
    <button
      className="DetailsCard--Collapser is-collapsed"
      onClick={[Function]}
      type="button"
    >
      <IoChevronDown />
    </button>
    <div
      className="DetailsCard--HeaderWrapper"
    >
      <span
        className="DetailsCard--Header"
      >
        Details Card Header
      </span>
    </div>
  </div>
  <div
    className="DetailsCard--DetailsWrapper is-collapsed"
  >
    <span />
  </div>
</div>
`;

exports[`DetailsCard renders list details 1`] = `
<div
  className="DetailsCard"
>
  <div
    className="DetailsCard--ButtonHeaderWrapper"
  >
    <div
      className="DetailsCard--HeaderWrapper"
    >
      <span
        className="DetailsCard--Header"
      >
        Details Card Header
      </span>
    </div>
  </div>
  <div
    className="DetailsCard--DetailsWrapper"
  >
    <DetailList
      details={
        Array [
          "foo",
          "bar",
          "baz",
        ]
      }
    />
  </div>
</div>
`;

exports[`DetailsCard renders string details 1`] = `
<div
  className="DetailsCard"
>
  <div
    className="DetailsCard--ButtonHeaderWrapper"
  >
    <div
      className="DetailsCard--HeaderWrapper"
    >
      <span
        className="DetailsCard--Header"
      >
        Details Card Header
      </span>
    </div>
  </div>
  <div
    className="DetailsCard--DetailsWrapper"
  >
    <span>
      test details
    </span>
  </div>
</div>
`;

exports[`DetailsCard renders table details 1`] = `
<div
  className="DetailsCard"
>
  <div
    className="DetailsCard--ButtonHeaderWrapper"
  >
    <div
      className="DetailsCard--HeaderWrapper"
    >
      <span
        className="DetailsCard--Header"
      >
        Details Card Header
      </span>
    </div>
  </div>
  <div
    className="DetailsCard--DetailsWrapper"
  >
    <DetailTable
      details={
        Array [
          Object {
            "value": "foo",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`DetailsCard renders table details with column defs 1`] = `
<div
  className="DetailsCard"
>
  <div
    className="DetailsCard--ButtonHeaderWrapper"
  >
    <div
      className="DetailsCard--HeaderWrapper"
    >
      <span
        className="DetailsCard--Header"
      >
        Details Card Header
      </span>
    </div>
  </div>
  <div
    className="DetailsCard--DetailsWrapper"
  >
    <DetailTable
      columnDefs={
        Array [
          "col",
        ]
      }
      details={
        Array [
          Object {
            "col": "foo",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`DetailsCard renders with className 1`] = `
<div
  className="DetailsCard test className"
>
  <div
    className="DetailsCard--ButtonHeaderWrapper"
  >
    <div
      className="DetailsCard--HeaderWrapper"
    >
      <span
        className="DetailsCard--Header"
      >
        Details Card Header
      </span>
    </div>
  </div>
  <div
    className="DetailsCard--DetailsWrapper"
  >
    <span />
  </div>
</div>
`;

exports[`DetailsCard renders with description 1`] = `
<div
  className="DetailsCard"
>
  <div
    className="DetailsCard--ButtonHeaderWrapper"
  >
    <div
      className="DetailsCard--HeaderWrapper"
    >
      <span
        className="DetailsCard--Header"
      >
        Details Card Header
      </span>
      <p
        className="DetailsCard--Description"
      >
        test description
      </p>
    </div>
  </div>
  <div
    className="DetailsCard--DetailsWrapper"
  >
    <span />
  </div>
</div>
`;
